### PR TITLE
Document containersPath and secretPath behavior

### LINKED ISSUE #641: Document containersPath and secretPath behavior
PR #513 introduced containersPath and secretPath. Documenting the behavior of its usage will be helpful to the user.

### PR DESCRIPTION
**Pull request by baijum**
Fixes #641

### Motivation

This is an advanced feature which required documentation.

### Changes

Add a new document file.

### Testing

Follow the instruction in the docs and make sure it works on a cluster.

For further more details refer the CONTRIBUTING.md

### CODE REVIEW

File: docs/binding-path.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Custom Containers Path and Secret Path
+
+## Containers Path
+
+If your application is using a custom resource and containers path should bind
+at a certain location, SBO provides an API to achieve that.  Here is an example
```
Comment (DhritiShikhar) [inline docs/binding-path.md:6]:
```suggestion
at a custom location, SBO provides an API to achieve that.  Here is an example
```

File: docs/binding-path.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    name: hello-world
+    ports:
+    - containerPort: 8090
+    resources: {}
+```
+
+## Secret Path
+
+If your application is using a custom resource and secret path should bind at a
+certain location, SBO provides an API to achieve that.  Here is an example CR
```
Comment (DhritiShikhar) [inline docs/binding-path.md:75]:
```suggestion
custom location, SBO provides an API to achieve that.  Here is an example CR
```

**Comment by qibobo** [comment]
@baijum @isutton 
I just realized that the PR 513, the bindingPath format has finally changed from `bindingPath.podSpecPath.containers` to `bindingPath.containersPath`, any special reason to change it? Actually we have picked the old custom binding path PR into our environments, so we need to use the old format as below.
```
    bindingPath:
      podSpecPath:
        containers: spec.spec.containers
```

**Comment by sbose78** [comment]
Sorry this change wasn't more explicitly communicated, but we could follow the discussion in PR #513 , there will be a long trail of comments on this.

> I just realized that the PR 513, the bindingPath format has finally changed from bindingPath.podSpecPath.containers to bindingPath.containersPath, any special reason to change it? 

It was separated out since we are no longer targeting a custom `podSpec` path - a custom `podspec` path might need specification of volumes as well. 

In the new API, one can have a custom containers path without the volumes path being necessary at all.

**Comment by pedjak** [comment]
/lgtm 

when @DhritiShikhar comments are adddressed

**Comment by baijum** [comment]
> /lgtm
> 
> when @DhritiShikhar comments are adddressed

Done.

Please add `/lgtm` and `/approve` comments again. Thanks!

**Pull request merged**